src/agni/client/HeartbeatMonitor.java

```diff
@@ -0,0 +1,39 @@
+package agni.client;
+
+public class HeartbeatMonitor implements Runnable {
+
+    private long TIMEOUT_PERIOD;
+
+    private long lastHeartbeatTime;
+    private boolean timedOut;
+
+    public HeartbeatMonitor(long timeoutPeriod) {
+        lastHeartbeatTime = System.nanoTime();
+        timedOut = false;
+        TIMEOUT_PERIOD = timeoutPeriod;
+    }
+
+    public void receivedHeartbeat() {
+        lastHeartbeatTime = System.nanoTime();
+    }
+
+    public final boolean isTimedOut() {
+        return timedOut;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
```

Comment:
Functionally they are the same, but the for(;;) seems more deliberate. But yea, while(true) came first in mind here, but I should've used `for(;;)` in my own advice xD